### PR TITLE
Add Custom Provider .Net SDK module

### PR DESCRIPTION
**Pull request by garyke**
Creating new SDK project for Azure Custom Provider RP and adding new tests

### CODE REVIEW

**Comment by dsgouda** [comment]
@garyke Please link the REST spec PR

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	<!-- Please do not move/edit code below this line -->
+
+	<PropertyGroup>
+		<Description>Provides developers with libraries to create and manage custom provider resource. Note: This client library is for CustomProviders under Azure Resource Manager.</Description>
+		<Version>0.1.1-preview</Version>
+		<AssemblyName>Microsoft.CustomProviders</AssemblyName>
+		<PackageId>Microsoft.CustomProviders</PackageId>
+		<PackageTags>CustomProviders;</PackageTags>
+		<PackageReleaseNotes />
+    <GenerateResourceNeverLockTypeAssemblies>true</GenerateResourceNeverLockTypeAssemblies>
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Please remove this line

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	</PropertyGroup>
+	<PropertyGroup>
+		<TargetFrameworks>$(SdkTargetFx)</TargetFrameworks>
+	</PropertyGroup>
+	<ItemGroup>
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Please remove this ItemGroup

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<!-- Please do not move/edit code below this line -->
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+	<!-- Please do not move/edit code below this line -->
+
+	<PropertyGroup>
+		<Description>Provides developers with libraries to create and manage custom provider resource. Note: This client library is for CustomProviders under Azure Resource Manager.</Description>
+		<Version>0.1.1-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Please start versioning from `0.8.0-preview`

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		<AssemblyName>Microsoft.CustomProviders</AssemblyName>
+		<PackageId>Microsoft.CustomProviders</PackageId>
+		<PackageTags>CustomProviders;</PackageTags>
+		<PackageReleaseNotes />
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Please add the `PackageReleaseNotes`

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<!-- Please do not move/edit code below this line -->
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+	<!-- Please do not move/edit code below this line -->
+
+	<PropertyGroup>
+		<Description>Provides developers with libraries to create and manage custom provider resource. Note: This client library is for CustomProviders under Azure Resource Manager.</Description>
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Please move description to `PackageReleaseNotes`

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj [hunk 6]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<!-- Please do not move/edit code below this line -->
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Please rename this directory to `Management.CustomProviders` and rename this file to `Microsoft.Azure.Management.CustomProviders.csproj`

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj (hunk 3 above)
Comment (garyke) [reply src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Using 0.8.1-preview

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj (hunk 5 above)
Comment (garyke) [reply src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Moved to PackageReleaseNotes

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj (hunk 1 above)
Comment (garyke) [reply src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: removed

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj (hunk 2 above)
Comment (garyke) [reply src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: removed

File: src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj (hunk 4 above)
Comment (garyke) [reply src/SDKs/CustomProviders/Microsoft.CustomProviders/Microsoft.CustomProviders.csproj]: Added description here

File: src/SDKs/CustomProviders/CustomProviders.Tests/CustomProviders.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.CustomProviders\Microsoft.CustomProviders.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+
+  <ItemGroup>
```
[the remark this answers is not in the document]
Comment (garyke) [reply src/SDKs/CustomProviders/CustomProviders.Tests/CustomProviders.Tests.csproj]: consolidated into less itemgroup

**Comment by dsgouda** [comment]
@garyke please push your new commits to the remote

**Comment by garyke** [comment]
@dsgouda Here's the spec link - https://github.com/Azure/azure-rest-api-specs/blob/master/specification/customproviders/resource-manager/Microsoft.CustomProviders/preview/2018-09-01-preview/customproviders.json

File: src/SDKs/CustomProviders/Management.CustomProviders/Microsoft.Azure.Management.CustomProviders.csproj [hunk 8]
```diff
@@ -4,20 +4,20 @@
 	<!-- Please do not move/edit code below this line -->
 
 	<PropertyGroup>
-		<Description>Provides developers with libraries to create and manage custom provider resource. Note: This client library is for CustomProviders under Azure Resource Manager.</Description>
-		<Version>0.1.1-preview</Version>
+		<Version>0.8.1-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Management.CustomProviders/Microsoft.Azure.Management.CustomProviders.csproj]: Please set this to `0.8.0-preview` instead

File: src/SDKs/CustomProviders/Management.CustomProviders/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Custom Providers Library")]
 [assembly: AssemblyDescription("Provides management functions for managing the Microsoft Azure CustomProviders service.")]
 
-[assembly: AssemblyVersion("0.1.1.0")]
-[assembly: AssemblyFileVersion("0.1.1.0")]
+[assembly: AssemblyVersion("0.8.1.0")]
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Management.CustomProviders/Properties/AssemblyInfo.cs]: Please set this to `0.8.0.0`

File: src/SDKs/CustomProviders/Management.CustomProviders/Microsoft.Azure.Management.CustomProviders.csproj [hunk 10]
```diff
@@ -4,20 +4,20 @@
...
 		<AssemblyName>Microsoft.CustomProviders</AssemblyName>
 		<PackageId>Microsoft.CustomProviders</PackageId>
 		<PackageTags>CustomProviders;</PackageTags>
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Management.CustomProviders/Microsoft.Azure.Management.CustomProviders.csproj]: You might want to add some additional tags like Azure and Microsoft.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a few comments

File: src/SDKs/CustomProviders/Management.CustomProviders/Properties/AssemblyInfo.cs [hunk 11]
```diff
@@ -7,7 +7,7 @@
 [assembly: AssemblyTitle("Microsoft Azure Custom Providers Library")]
 [assembly: AssemblyDescription("Provides management functions for managing the Microsoft Azure CustomProviders service.")]
 
-[assembly: AssemblyVersion("0.8.1.0")]
+[assembly: AssemblyVersion("0.8.0.0")]
 [assembly: AssemblyFileVersion("0.8.1.0")]
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Management.CustomProviders/Properties/AssemblyInfo.cs]: Here too

**Review by dsgouda — APPROVED** [review]
LGTM will merge on CIs passing

File: src/SDKs/CustomProviders/Management.CustomProviders/generate.ps1 [hunk 12]
```diff
@@ -0,0 +1 @@
+Start-AutoRestCodeGeneration -ResourceProvider "customproviders/resource-manager" -AutoRestVersion "latest"
```
Comment (dsgouda) [inline src/SDKs/CustomProviders/Management.CustomProviders/generate.ps1]: Was the code generated using this file? If yes, it should have created a .txt file, please commit this file.

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**